src/r-apt/install.sh

```diff
@@ -123,8 +126,20 @@ Pin: release l=CRAN-Apt Packages
 Pin-Priority: 700
 EOF
 elif [ "${ID}" = "debian" ]; then
-    curl -fsSL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x95c0faf38db3ccad0c080a7bdc78b2ddeabc47b7" | tee -a /etc/apt/trusted.gpg.d/cran_debian_key.asc
-    echo "deb [arch=amd64] http://cloud.r-project.org/bin/linux/debian ${VERSION_CODENAME}-cran40/" >/etc/apt/sources.list.d/cran-debian.list
+    if [ "${USE_UNSTABLE}" = "true" ]; then
+        echo "Set up Debian unstable..."
+        echo "deb http://http.debian.net/debian sid main" >/etc/apt/sources.list.d/debian-unstable.list
+        # Pinning
+        cat <<EOF >"/etc/apt/preferences.d/99debian-unstable"
+Package: *
+Pin: release a=unstable
+Pin-Priority: 700
+EOF
+    else
+        echo "Set up for Debian ${VERSION_CODENAME}..."
+        curl -fsSL "https://keyserver.ubuntu.com/pks/lookup?op=get&search=0x95c0faf38db3ccad0c080a7bdc78b2ddeabc47b7" | tee -a /etc/apt/trusted.gpg.d/cran_debian_key.asc >/dev/null
+        echo "deb [arch=amd64] http://cloud.r-project.org/bin/linux/debian ${VERSION_CODENAME}-cran40/" >/etc/apt/sources.list.d/cran-debian.list
+    fi
```

Comment:
Are we sure this helps and works and is useful?  The older release are mostly for 'stability'. You are building a container here so there is little risk of "spilling over" onto a main system.

Reply:
I think you are right.
I think installing the old R packages are just an inconvenience most of the time. I am just implementing the way it was documented in the CRAN documentation.

Perhaps the default behavior is not to use this.

Reply:
Also see how `r-base` defaults to testing while allowing unstable:

https://github.com/rocker-org/rocker/blob/53be7be0a1ab52534162910b4ab48f597d920664/r-base/latest/Dockerfile#L38-L41

Reply:
Of course I checked that, but I mistakenly thought it was a setup for installing R packages from unstable since the base image is already testing.
Why does r-base also set up unstable?

Reply:
See https://github.com/rocker-org/devcontainer-features/pull/86#discussion_r1008699711

Reply:
A concrete example:   Sometimes right before a release Debian stabilizes and packages do not transition from unstable to testing in the usual automated fashion.  In the cases I pick the R versions I want for r-base explicitly from unstable.  And sometimes you just want to pick a package or two from unstable.  Using the above setup with `apt install -t unstable foo bar` allows that while defaulting to testing for the normal use.  Best of both worlds.

Reply:
Thanks for the detailed explanation, given that r-base has a good enough track record, it seems logical to set up unstable as well as r-base.